addons/account/models/account_move.py

```diff
@@ -698,6 +698,14 @@ def default_get(self, fields):
             rec.update({'credit': balance})
         return rec
 
+    @api.multi
+    @api.constrains('credit', 'debit', 'tax_ids', 'tax_line_id', 'date')
```

Comment:
that ain't working... 

the implementation with a constraint will fail in the following use case:
- make an invoice with a tax
- set the tax lock date to include the invoice
- go on the invoice journal entry and remove the tax_ids or tax_line_id values (with account_cancel, or via the studio)
=> an error should be thrown but there isn't any since the test at line 705 is falsy

Reply:
I don't get how it is different from what is done in the tests? And the tests are green.

Reply:
because in my use case i'm modifying one of the field that composes the condition of L705, making it False where it was True before.

please, read again my use case

Reply:
Oh, right... I remember having thought about this though.
Updated, with the tests too.